drivers/MbedCRC.cpp

```diff
@@ -22,6 +22,8 @@ namespace mbed {
 /** \addtogroup drivers */
 /** @{*/
 
+SingletonPtr<PlatformMutex> mbed_crc_mutex;
```

Comment:
Locking is needed for HW, since we will have one HW module performing CRC computation for all the instances. I understand this is an overhead for software only computations or combination of software and hardware. 

We can have lock/unlock only in case of hardware, does that sound correct?